### PR TITLE
pytest for test_db.py

### PR DESCRIPTION
**Pull request by heartsucker**
## Status

Ready for review

## Description of Changes

Towards #2877.

Moved `test_db.py` to pytest fixtures.

## Testing

```
make tests TESTFILES=tests/test_db.py
```

### If you made changes to the server application code:

- [x] Linting (`make ci-lint`) and tests (`make -C securedrop test`) pass in the development container

### CODE REVIEW

File: securedrop/tests/test_db.py [hunk 1]
```diff
@@ -56,6 +56,15 @@ def test_get_one_or_else_no_result_found(journalist_app, test_journo):
         mock_abort.assert_called_with(404)
 
 
+def test_submission_string_representation(journalist_app):
+    with journalist_app.app_context():
+        source, _ = db_helper.init_source()
```
Comment (redshiftzero) [inline securedrop/tests/test_db.py]: One could use the `test_source` fixture to remove this line 😇

File: securedrop/tests/test_db.py [hunk 2]
```diff
@@ -65,6 +65,15 @@ def test_submission_string_representation(journalist_app):
         test_submission.__repr__()
 
 
+def test_reply_string_representation(journalist_app):
+    with journalist_app.app_context():
+        journalist, _ = db_helper.init_journalist()
+        source, _ = db_helper.init_source()
```
Comment (redshiftzero) [inline securedrop/tests/test_db.py]: Again, we could use the `test_journalist` and `test_source` fixtures here

File: securedrop/tests/test_db.py [hunk 3]
```diff
@@ -74,6 +74,13 @@ def test_reply_string_representation(journalist_app):
         test_reply.__repr__()
 
 
+def test_journalist_string_representation(journalist_app):
+    with journalist_app.app_context():
+        test_journalist, _ = db_helper.init_journalist()
```
Comment (redshiftzero) [inline securedrop/tests/test_db.py]: `test_journalist` fixture

File: securedrop/tests/test_db.py [hunk 4]
```diff
@@ -80,6 +80,11 @@ def test_journalist_string_representation(journalist_app):
         test_journalist.__repr__()
 
 
+def test_source_string_representation(journalist_app):
+    with journalist_app.app_context():
+        test_source, _ = db_helper.init_source()
```
Comment (redshiftzero) [inline securedrop/tests/test_db.py]: `test_source` fixture here

File: securedrop/tests/test_db.py [hunk 5]
```diff
@@ -56,6 +55,15 @@ def test_get_one_or_else_no_result_found(journalist_app, test_journo):
         mock_abort.assert_called_with(404)
 
 
+def test_throttle_login(journalist_app):
+    with journalist_app.app_context():
+        journalist, _ = db_helper.init_journalist()
```
Comment (redshiftzero) [inline securedrop/tests/test_db.py]: And the `test_journalist` fixture here

**Comment by heartsucker** [comment]
@redshiftzero Rebased and pushed. I thought that SqlAlchemy would whine if you detached the session from an object, kept the ref around, then reattached the session. Turns out this only matters if you attempt to access the object while the session is detached, so added the objects to the fixture is fine. TIL.

**Pull request merged**